src/apps/FindMySafe/index.tsx

```diff
@@ -0,0 +1,299 @@
+  }
+`;
+
+const SafeBalances = styled.div`
+  margin-top: 55px;
+  width: 100%;
+`;
+
+const NoBalanceFound = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100%;
+`;
+
+const IconImage = styled.div`
+  display: flex;
+  align-items: center;
+
+  img {
+    margin-right: 5px;
+  }
+`;
+
+const BalancesContainer = styled.div`
+  height: 230px;
+  width: 400px;
+  overflow-y: auto;
+`;
+
+type BalanceInfo = {
+  tokenAddress: string;
+  token: {
+    name: string;
+    symbol: string;
+    decimals: number;
+    logoUri: string;
+  };
+  balance: string;
+};
+
+const getSafes = memoize(
+  async (address: string, network: Networks): Promise<string[]> => {
+    const res = await axios.get<{ safes: string[] }>(
+      `${apiNetwork[network]}/owners/${address}/`
+    );
+    return res.data.safes;
+  }
+);
+
+const getBalances = memoize(
+  async (safe: string, network: Networks): Promise<BalanceInfo[]> => {
+    const res = await axios.get<BalanceInfo[]>(
+      `${apiNetwork[network]}/safes/${safe}/balances/usd/`
+    );
+
+    return res.data.filter((b) => b.token !== null);
+  }
+);
+
+export const setImageToPlaceholder: React.ReactEventHandler<HTMLImageElement> = (
+  e
+) => {
+  (e.target as HTMLImageElement).onerror = null;
+  (e.target as HTMLImageElement).src = TokenPlaceholder;
+};
+
+const FindMySafe = () => {
+  const [appsSdk] = useState(initSdk());
+  const [safeInfo, setSafeInfo] = useState<SafeInfo>();
+
+  const [address, setAddress] = useState("");
+  const [safes, setSafes] = useState<string[]>([]);
+  const [loadingSafes, setLoadingSafes] = useState(false);
+  const [loadingBalances, setLoadingBalances] = useState(false);
+  const [selectedSafe, setSelectedSafe] = useState<string | undefined>();
+  const [balances, setBalances] = useState<BalanceInfo[] | undefined>([]);
+
+  // config safe connector
+  useEffect(() => {
+    appsSdk.addListeners({
+      onSafeInfo: setSafeInfo,
+    });
+
+    return () => appsSdk.removeListeners();
+  }, [appsSdk]);
+
+  const handleSearch = async (
+    event: React.ChangeEvent<HTMLInputElement>
+  ): Promise<void> => {
+    const address = event.target.value;
+    setAddress(address);
+    setSelectedSafe(undefined);
+    setBalances(undefined);
+
+    if (!address.length) {
+      setSafes([]);
+      setSelectedSafe(undefined);
+      return;
+    }
+
+    setLoadingSafes(true);
+    try {
+      const res = await getSafes(address, safeInfo!.network);
+      setSafes(res);
+      setLoadingSafes(false);
+    } catch (error) {
+      setLoadingSafes(false);
+    }
+  };
+
+  const handleSafeClick = async (safe: string): Promise<void> => {
+    setSelectedSafe(safe);
+    setLoadingBalances(true);
+
+    try {
+      const res = await getBalances(safe, safeInfo!.network);
+      setBalances(res);
+      setLoadingBalances(false);
+    } catch (error) {
+      setLoadingBalances(false);
+    }
+  };
+
+  const getBalancesContent = () => {
```

Comment:
this file is getting too big and hard to follow, is it possible to split it into separate components? for example, you can extract this to a separate component `Balances`

Reply:
Sorry, wasn't ready to review yet. I created it to test something and I forget to put it on WIP

Reply:
@mikheevm ready for review now!